src/tensor_operations/matrix_algebra.jl

```diff
@@ -6,7 +6,10 @@ function _tr(T::ITensor; plev::Pair{Int,Int}=0 => 1, tags::Pair=ts"" => ts"")
   Tᶜ = T * Cᴸ * Cᴿ
   cᴸ = uniqueind(Cᴸ, T)
   cᴿ = uniqueind(Cᴿ, T)
-  Tᶜ *= δ(dag((cᴸ, cᴿ)))
+  ## deafult eltype for delta is Float64 so this 
+  ## can promote eltype from F32 to F64
+  elt = eltype(T)
+  Tᶜ *= δ(elt, dag((cᴸ, cᴿ)))
```

Comment:
```suggestion
  Tᶜ *= δ(eltype(T), dag((cᴸ, cᴿ)))
```

Reply:
I think we can keep it simple, it's pretty self-explanatory.

Reply:
Sounds good to me, thanks!